docs/tools/install.sh

```diff
@@ -0,0 +1,228 @@
+        return 1
+    fi
+
+    exists=$(kubectl get directpvvolumes.directpv.min.io --selector=directpv.min.io/migrated=true --ignore-not-found=true -o go-template='{{range .items}}{{1}}{{break}}{{end}}')
+    [ -n "${exists}" ]
+}
+
+declare -A fsuuidDriveNameMap driveNameFSUUIDMap
+
+function check_directcsi_consistency() {
+    ecount=0
+    # shellcheck disable=SC2207
+    names=( $(kubectl get directcsidrives -o go-template='{{range .items}}{{if or (eq .status.driveStatus "Ready") (eq .status.driveStatus "InUse") }}{{.metadata.name}} {{end}}{{end}}') )
+    for name in "${names[@]}"; do
+        fsuuid=$(kubectl get directcsidrives "${name}" -o go-template='{{.status.filesystemUUID}}')
+
+        if [ -z "${fsuuid}" ]; then
+            echo "[ERROR] drive ${name}: empty fsuuid"
+            (( ++ecount ))
+            continue
+        fi
+
+        if ! echo "${fsuuid}" | grep -q -E "^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$"; then
+            echo "[ERROR] drive ${name}: invalid fsuuid ${fsuuid}"
+            (( ++ecount ))
+            continue
+        fi
+
+        if [ -n "${fsuuidDriveNameMap[$fsuuid]}" ]; then
+            echo "[ERROR] drive ${name}: duplicate FSUUID ${fsuuid} found"
+            (( ++ecount ))
+            continue
+        fi
+
+        fsuuidDriveNameMap[$fsuuid]=$name
+        driveNameFSUUIDMap[$name]=$fsuuid
+    done
+
+    if [ "${ecount}" -gt 0 ]; then
+        exit 1
+    fi
+
+    for nameDrive in $(kubectl get directcsivolumes -o go-template='{{range .items}}{{.metadata.name}}={{.status.drive}} {{end}}'); do
+        name=$(echo "${nameDrive}" | cut -d= -f1)
+        drive=$(echo "${nameDrive}" | cut -d= -f2)
+
+        if [ -z "${drive}" ]; then
+            echo "[ERROR] volume ${name}: empty drive name"
+            (( ++ecount ))
+            continue
+        fi
+
+        if [ -z "${driveNameFSUUIDMap[$drive]}" ]; then
+            echo "[ERROR] volume ${name}: FSUUID not found for drive ${drive}"
+            (( ++ecount ))
+            continue
+        fi
+    done
+
+    if [ "${ecount}" -gt 0 ]; then
+        exit 1
+    fi
+}
+
+function init() {
+    help_flag=0
+    for arg in "$@"; do
+        if [ "${arg}" != "apply" ]; then
+            directpv_args+=("${arg}")
+            if [ "${arg}" == "--help" ]; then
+                help_flag=1
+            fi
+        else
+            apply_flag=1
+        fi
+    done
+
+    if [ "${help_flag}" -eq 1 ]; then
+        cat << EOF
+USAGE:
+  install.sh [INSTALL-FLAGS] [apply]
+
+ARGUMENTS:
+  INSTALL-FLAGS    Optional DirectPV installation flags.
+  apply            If present, apply manifest file after generation.
+
+EXAMPLES:
+  # Generate DirectPV manifests.
+  $ install.sh
+
+  # Generate DirectPV manifests with private registry.
+  $ install.sh --registry private-registry.io --org org-name
+
+  # Generate DirectPV manifests with node-selector.
+  $ install.sh --node-selector node-label-key=node-label-value
```

Comment:
Is there a way to test local dev images? can we also add `--image`?

Reply:
You can pass whatever `kubectl-directpv install` command has. These examples are to indicate how to use `install` command's arguments.

Reply:
this is not an issue also, we can edit the image manually in the manifest. just that we can't use the script

Reply:
Your expectation is wrong. The image you are expecting has to be in the registry or in the local registry. It is a bad idea to modify the manifests here.

Reply:
I meant we can do `kubectl directpv install --declarative > manifest.yaml` and then editing the `image` field to our dev image.

I see a point that `dev` images will not be of real use here. So, there is not need to use this script for development.